# Домашнее задание № 3

## по дисциплине: Теория вероятностей и математическая статистика

## Тема: Основы математической статистики. Количественные характеристики популяции. Графическое представление данных

### выполнил: Евгений Кириллов

__Задача 1__


Даны значения зарплат из выборки выпускников:

100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 230, 24, 57, 55, 70,
75, 65, 84, 90, 150

Используя только встроенные питоновские функции и структуры
данных (т.е. без библиотек numpy, pandas и др.), посчитайте
(несмещённое) среднее квадратичное отклонение для данной
выборки.

Ответ: 47.34795214

In [1]:
salaries = [100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 230, 24, 57, 55, 70, 75, 65, 84, 90, 150]

посчитаем выборочное среднее:

$$\overline{X} = \dfrac{1}{n} \displaystyle\sum_{i = 1}^n x_i$$

In [2]:
salaries_mean = sum(salaries) / len(salaries)
salaries_mean

73.14285714285714

посчитаем несмещённую выборочную дисперсию:

$$\sigma_X^2 = \dfrac{1}{n - 1} \displaystyle\sum_{i = 1}^n \left( x_i - \overline{X} \right)^2$$

In [3]:
salaries_varience = sum([(i - salaries_mean) ** 2 for i in salaries]) / (len(salaries) - 1)
salaries_varience

2241.8285714285716

посчитаем несмещённое среднее квадратическое отклонение:

$$\sigma_X = \sqrt{\dfrac{1}{n - 1} \displaystyle\sum_{i = 1}^n \left( x_i - \overline{X} \right)^2}$$

In [4]:
salaries_std = salaries_varience ** 0.5
salaries_std

47.34795213553139

ответ: 47.34795213553139

__Задача 2__


Найдите число выбросов в выборке из задачи 1. Для определения
выбросов используйте методику как при построении “усов” в boxplot,
однако, как и в задаче 1, пользоваться можно только встроенными
функциями и структурами данных.


Ответ: 2

In [5]:
len(salaries)

21

In [6]:
sorted(salaries)

[17,
 24,
 25,
 30,
 33,
 45,
 55,
 57,
 65,
 65,
 70,
 75,
 75,
 77,
 80,
 84,
 89,
 90,
 100,
 150,
 230]

найдём Q1

In [7]:
q1 = sorted(salaries)[len(salaries) // 4]
q1

45

найдём Q3

In [8]:
q3 = sorted(salaries)[3 * len(salaries) // 4]
q3

84

найдём интерквартильное расстояние IQR:

In [9]:
iqr = q3 - q1
iqr

39

вычислим диапазон ящика с усами boxplot_range:

In [10]:
boxplot_range = (q1 - 1.5 * iqr, q3 + 1.5 * iqr)
boxplot_range

(-13.5, 142.5)

найдём выбросы и посчитаем их количество:

In [11]:
outliers = [i for i in salaries if i < boxplot_range[0] or i > boxplot_range[1]]
outliers

[230, 150]

In [12]:
len(outliers)

2

ответ: 2

__Задача 3__


В университет на факультет A поступило 100 человек, на факультет B - 90 человек и на факультет C - 45 человек. Вероятность того, что
студент с факультета A сдаст первую сессию, равна 0.6. Для студента
с факультета B эта вероятность равна 0.7, а для студента с
факультета C - 0.8. Случайным образом выбирается студент, сдавший
первую сессию. Какое событие наиболее вероятно:
1. студент учится на факультете A,
2. студент учится на факультете B,
3. студент учится на факультете C?


Замечание: да, эта задача не на тему статистики, но тема важная, и её
стоит иногда освежать в памяти.


Ответ: 2

In [13]:
number_of_all_students = 100 + 90 + 45
number_of_all_students

235

In [14]:
bayes_full_probability = (100 / number_of_all_students) * 0.6 +\
    (90 / number_of_all_students) * 0.7 + (45 / number_of_all_students) * 0.8
bayes_full_probability

0.676595744680851

In [15]:
event_1 = (100 / number_of_all_students) * 0.6 / bayes_full_probability
event_1

0.3773584905660377

In [16]:
event_2 = (90 / number_of_all_students) * 0.7 / bayes_full_probability
event_2

0.39622641509433965

In [17]:
event_3 = (45 / number_of_all_students) * 0.8 / bayes_full_probability
event_3

0.22641509433962265

ответ: 2